File: src/sorting/bubbleSort.cs
using System;
using Sorting;

namespace BubbleSort
{
    public class BubbleSort<T> : Sorting<T> where T : IComparable<T>
    {
        public BubbleSort()
        {
            _arr = Array.ConvertAll<string, T>(Console.ReadLine().Split(' ',
            StringSplitOptions.RemoveEmptyEntries), ParseType);
            _count = _arr.Length;
        }

        public BubbleSort(T[] input)
        {
            _arr = input;
            _count = _arr.Length;
        }

        public override void Sort()
        {
            for (int i = 0; i < _count - 1; i++)
            {
                for (int j = 0; j < _count - 1 - i; j++)
                    if (_arr[j].CompareTo(_arr[j + 1]) > 0) // _arr[j] is bigger than _arr[j + 1];
                        Swap(j, j + 1);
            }
        }
    }
    public static class Program
    {
        public static void Main()
        {
            var a = new BubbleSort<int>();
            a.PrintArray();
            a.Sort();
            a.PrintArray();
        }
    }
}

/*
Input:
5 4 3 2 1

Output:
5 4 3 2 1
1 2 3 4 5
*/

File: src/sorting/bubbleSort.py
def bubble_sort(arr: list, size: int) -> None:
    for i in range(0, size - 1):
        for j in range(0, size - 1 - i):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]


def print_array(arr: list, size: int) -> None:
    for i in range(0, size):
        print(arr[i], end=' ')
    print()


arr = list(map(int, input().split()))
size = len(arr)

print_array(arr, size)
bubble_sort(arr, size)
print_array(arr, size)

'''
Input:
5 4 3 2 1

Output:
5 4 3 2 1
1 2 3 4 5
'''